code/controllers/subsystem/garbage.dm

```diff
@@ -298,8 +299,10 @@ SUBSYSTEM_DEF(garbage)
 
 				SSgarbage.Queue(D)
 			if (QDEL_HINT_HARDDEL)		//qdel should assume this object won't gc, and queue a hard delete
+				SSdemo.mark_destroyed(D)
```

Comment:
Okay after trying to understand this myself, this is the one that needs to be removed
```suggestion
```

Reply:
No, queueing a HARDDELETE causes it to harddel immediately, not actually queue one, so after the Queue statement it *will* be null and the `mark_destroyed` call wont work because it's null.